crates/core/c8y_api/src/json_c8y_deserializer.rs

```diff
@@ -413,6 +424,77 @@ pub struct C8yFirmware {
     pub url: String,
 }
 
+#[derive(Debug, Deserialize, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct C8yDeviceProfile {
+    pub firmware: C8yFirmware,
+    pub software: C8ySoftwareUpdate,
+    pub configuration: Vec<C8yDownloadConfigFile>,
+}
+
+impl C8yDeviceProfile {
+    pub fn get_firmware_info(&self) -> FirmwareInfo {
+        FirmwareInfo {
+            name: Some(self.firmware.name.clone()),
+            version: Some(self.firmware.version.clone()),
+            remote_url: Some(self.firmware.url.clone()),
+        }
+    }
+
+    pub fn try_get_software_info(&self) -> Result<SoftwareInfo, C8yJsonOverMqttDeserializerError> {
+        let mut software_info = SoftwareInfo {
+            update_list: Vec::new(),
+        };
+
+        for module in self.software.modules() {
+            let plugin_type = module
+                .get_module_version_and_type()
+                .1
+                .unwrap_or_else(SoftwareModule::default_type);
+
+            let item = SoftwareModuleItem {
+                name: module.name.clone(),
+                version: module.get_module_version_and_type().0,
+                url: module.get_url(),
+                action: match module.action.clone().try_into()? {
+                    C8ySoftwareUpdateAction::Install => Some(SoftwareModuleAction::Install),
+                    C8ySoftwareUpdateAction::Delete => Some(SoftwareModuleAction::Remove),
+                },
+                reason: None,
+            };
+
+            if let Some(list) = software_info
+                .update_list
+                .iter_mut()
+                .find(|list| list.plugin_type == plugin_type)
+            {
+                list.modules.push(item);
+            } else {
+                software_info
+                    .update_list
+                    .push(SoftwareRequestResponseSoftwareList {
+                        plugin_type,
+                        modules: vec![item],
+                    });
+            }
+        }
+
+        Ok(software_info)
+    }
+
+    pub fn get_config_info(&self) -> Vec<ConfigInfo> {
+        let mut config_info = Vec::new();
+        for config in &self.configuration {
+            config_info.push(ConfigInfo {
+                config_type: config.config_type.clone(),
+                remote_url: Some(config.url.clone()),
+            })
+        }
```

Comment:
This part can be implemented as `From<C8yDownloadConfigFile> for ConfigInfo`.

Reply:
Resolved by 5efc5fb95343b4624fab8176b25e6a48dd9da66f